lib/discordrb/api.rb

```diff
@@ -97,6 +97,8 @@ def request(key, major_parameter, type, *attributes)
     # Specify RateLimit precision
     attributes.last[:x_ratelimit_precision] = 'millisecond'
 
+    # Force content_type to JSON if request is post/put
+    attributes.last[:content_type] = :json if %i[put post].include?(type)
```

Comment:
Unfortunately we cannot do this, as certain webhook and message based requests require the use of 'multipart/form-data`.